fiftyone/brain/internal/core/uniqueness.py

```diff
@@ -107,25 +117,38 @@ def compute_uniqueness(
     else:
         agg_fcn = None
 
-    embeddings, sample_ids, _ = fbu.get_embeddings(
-        samples,
-        model=model,
-        model_kwargs=model_kwargs,
-        patches_field=roi_field,
-        embeddings_field=embeddings_field,
-        embeddings=embeddings,
-        force_square=force_square,
-        alpha=alpha,
-        handle_missing="image",
-        agg_fcn=agg_fcn,
-        batch_size=batch_size,
-        num_workers=num_workers,
-        skip_failures=skip_failures,
-        progress=progress,
-    )
+    if similarity_index:
+        sample_ids = similarity_index.sample_ids
+        embeddings = np.array(
+            similarity_index.get_embeddings(sample_ids=sample_ids)[0]
+        )
+    elif config.similarity_method:
+        embeddings, sample_ids, _ = fbu.get_embeddings(
+            samples,
+            model=model,
+            model_kwargs=model_kwargs,
+            patches_field=roi_field,
+            embeddings_field=embeddings_field,
+            embeddings=embeddings,
+            force_square=force_square,
+            alpha=alpha,
+            handle_missing="image",
+            agg_fcn=agg_fcn,
+            batch_size=batch_size,
+            num_workers=num_workers,
+            skip_failures=skip_failures,
+            progress=progress,
+        )
+        brain_method = config.similarity_method.build()
```

Comment:
Lines 142-146 can be replaced with this:

```py
similarity_index = fob.compute_similarity(
    dataset,
    backend=similarity_backend,
    embeddings=embeddings,
)
```

which would allow you to remove lines 212-233 that you added, which are copy pasta from the `compute_similarity()` method. Best to let `compute_similarity()` handle all that logic.

One *additional* thing you could do is include `brain_key=similarity_index`:

```py
similarity_index = fob.compute_similarity(
    dataset,
    backend=similarity_backend,
    brain_key=similarity_index,
    embeddings=embeddings,
)
```

which would then allow the user to use this syntax:

```py
fob.compute_uniqueness(dataset, ..., similarity_backend="pinecone", similarity_index="img_sim", ...)
```

which says "please create a new Pinecone similarity index for this uniqueness computation and save it under `brain_key="img_sim"` so that I can reuse it later"

Reply:
Actually the *BEST* implementation would be this:

```py
similarity_index = fob.compute_similarity(
    samples,
    backend=similarity_backend,
    brain_key=similarity_index,
    embeddings=False,  # don't compute embeddings yet
)

embeddings, sample_ids, _ = fbu.get_embeddings(...)

similarity_index.add_to_index(embeddings, sample_ids)
```

The benefit of this is that if you call:

```py
fob.compute_uniqueness(dataset, similarity_backend="alskdjflaskdfj")
```

then you will immediately get a "similarity backend 'alskdjflaskdfj' not found" error.

In the current implementation, you'll spend a potentially long amount of time computing `embeddings` only to *then* realize you had a typo in your `similarity_backend`. I have personally refactored other methods in the past after users have complained that their typos were only surfaced *after* performing some hugely expensive compute

Reply:
Changed to the BEST option :)